crates/breez-sdk/core/src/sdk.rs

```diff
@@ -254,18 +257,21 @@ impl BreezSdk {
                             }
                         }
                     }
-                    () = async {
-                      let sync_type_res = sync_trigger_receiver.recv().await;
-                      if let Ok(sync_type) = sync_type_res   {
-                          info!("Sync trigger changed: {:?}", &sync_type);
-
-                          if let Err(e) = sdk.sync_wallet_internal(sync_type.clone()).await {
-                              error!("Failed to sync wallet: {e:?}");
-                          } else if matches!(sync_type, SyncType::Full) {
-                            last_sync_time = SystemTime::now();
-                          }
-                      }
-                  } => {}
+                    sync_type_res = sync_trigger_receiver.recv() => {
+                        if let Ok(sync_type) = sync_type_res   {
+                            info!("Sync trigger changed: {:?}", &sync_type);
+
+                            if let Err(e) = sdk.sync_wallet_internal(sync_type.clone()).await {
+                                error!("Failed to sync wallet: {e:?}");
+                            } else if matches!(sync_type, SyncType::Full) {
+                                last_sync_time = SystemTime::now();
+                            }
+
+                            if let Err(e) = sdk.sync_service.sync_historical_payments().await {
```

Comment:
I see sync_historical_payments is called anyway whenever the sync_wallet_internal is called. Perhaps it should be an implementation details as part of the `sync_service_sync_payments()` function?
Implemented this way only in sparkscan strategy?